packages/mock/src/msw/index.ts

```diff
@@ -129,3 +137,73 @@ export const ${handlerName} = (${isReturnHttpResponse && !isTextPlain ? `overrid
     imports: includeResponseImports,
   };
 };
+
+export const generateMSW = (
+  generatorVerbOptions: GeneratorVerbOptions,
+  generatorOptions: GeneratorOptions,
+) => {
+  const { pathRoute, override, mock } = generatorOptions;
+  const { operationId, response } = generatorVerbOptions;
+
+  const route = getRouteMSW(
+    pathRoute,
+    override?.mock?.baseUrl ?? (!isFunction(mock) ? mock?.baseUrl : undefined),
+  );
+
+  const handlerName = `get${pascal(operationId)}MockHandler`;
+  const getResponseMockFunctionName = `get${pascal(operationId)}ResponseMock`;
+
+  const baseDefinition = generateDefinition(
```

Comment:
There's a difference between the "base" and other success responses as I didn't want to introduce a breaking change. The "base" response behaves like the response generated before my change, which picks a random success response each time e.g.

```typescript
export const getListPetsResponseMock = (
  overrideResponse: any = {},
): Okay | Accepted | Success =>
  faker.helpers.arrayElement([ // pick a random "success" response
    Array.from( // the "Okay" type (an array of pets)
      { length: faker.number.int({ min: 1, max: 10 }) },
      (_, i) => i + 1,
    ).map(() =>
      faker.helpers.arrayElement([
        {
          breed: faker.helpers.arrayElement(['Labradoodle'] as const),
         ...
        },
        ...
      ]),
    ),
    { // The "Accepted" response
      code: faker.number.int({ min: undefined, max: undefined }),
      message: faker.word.sample(),
      ...overrideResponse,
    },
    { // The "Success" response
      code: faker.number.int({ min: undefined, max: undefined }),
      message: faker.word.sample(),
      ...overrideResponse,
    },
  ]);
```

Whereas the other mocks that are generated only return the specific response for a status:

```typescript
export const getListPetsResponseMock200 = (overrideResponse: any = {}): Okay =>
  Array.from(
    { length: faker.number.int({ min: 1, max: 10 }) },
    (_, i) => i + 1,
  ).map(() =>
    faker.helpers.arrayElement([
      {
        breed: faker.helpers.arrayElement(['Labradoodle'] as const),
        ...
      },
      ...
    ]),
  );

export const getListPetsResponseMock202 = (
  overrideResponse: any = {},
): Accepted => ({
  code: faker.number.int({ min: undefined, max: undefined }),
  message: faker.word.sample(),
  ...overrideResponse,
});

export const getListPetsResponseMock2xx = (
  overrideResponse: any = {},
): Success => ({
  code: faker.number.int({ min: undefined, max: undefined }),
  message: faker.word.sample(),
  ...overrideResponse,
});
```

I could change it to return a single success response for the "base" response if you would be happy with a breaking change, though it could be difficult to know which response to return (I guess it could be the lowest 2xx number defined in the responses).

Reply:
@huwshimi 

that's sounds good. I had missed this.
In other words, does this mean that when multiple success responses are defined, the mocked types are different, so it is not a duplicate, right?

Reply:
> @huwshimi
> 
> that's sounds good. I had missed this. In other words, does this mean that when multiple success responses are defined, the mocked types are different, so it is not a duplicate, right?

That's correct.